app/filtering.js

```diff
@@ -561,11 +561,24 @@ function initForPartition (partition) {
     registerForDownloadListener,
     registerForMagnetHandler]
   let options = {}
+
   if (isSessionPartition(partition)) {
     options.parent_partition = ''
   }
+
   let ses = session.fromPartition(partition, options)
-  fns.forEach((fn) => { fn(ses, partition, module.exports.isPrivate(partition)) })
+  fns.forEach((fn) => {
+    fn(ses, partition, module.exports.isPrivate(partition))
+  })
+  ses.on('register-navigator-handler', (e, protocol, location) => {
+    appActions.navigatorHandler(ses.partition, protocol, location, true)
```

Comment:
Good call 👍  Since this is already merged, please feel free to do as a follow up, @NejcZdovc 😄